tests/unit/test_state.py

```diff
@@ -208,11 +259,150 @@ def test_config_protocol_invalid():
+
+
+def test_non_int_time_proxy_cache_valid():
+    """
+    arrange: Mock charm with cache valid config with non int time.
+    act: Create the state from the charm.
+    assert: Configuration error raised with a correct error message.
+    """
+    charm = MockCharmFactory()
+    charm.config[PROXY_CACHE_VALID_CONFIG_NAME] = '["200 302 tend"]'
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+
+    assert "Value error, Non-int time in proxy_cache_valid: tend" in str(err.value)
+
+
+def test_negative_time_proxy_cache_valid():
+    """
+    arrange: Mock charm with cache valid config with negative time.
+    act: Create the state from the charm.
+    assert: Configuration error raised with a correct error message.
+    """
+    charm = MockCharmFactory()
+    charm.config[PROXY_CACHE_VALID_CONFIG_NAME] = '["200 302 -10d"]'
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+
+    assert "Value error, Time must be positive int for proxy_cache_valid: -10d" in str(err.value)
+
+
+def test_non_int_status_code_proxy_cache_valid():
+    """
+    arrange: Mock charm with cache valid config with non-int status code.
+    act: Create the state from the charm.
+    assert: Configuration error raised with a correct error message.
+    """
+    charm = MockCharmFactory()
+    charm.config[PROXY_CACHE_VALID_CONFIG_NAME] = '["ok 30m"]'
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+
+    assert "Value error, Non-int status code in proxy_cache_valid: ok" in str(err.value)
+
+
+def test_invalid_status_code_proxy_cache_valid():
+    """
+    arrange: Mock charm with cache valid config with invalid status code.
+    act: Create the state from the charm.
+    assert: Configuration error raised with a correct error message.
+    """
+    charm = MockCharmFactory()
+    charm.config[PROXY_CACHE_VALID_CONFIG_NAME] = '["200 99 30m"]'
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+
+    assert "Value error, Invalid status code in proxy_cache_valid: 99" in str(err.value)
+
+
+def test_valid_proxy_cache_valid():
+    """
+    arrange: Mock charm with valid complex proxy_cache_valid configuration.
+    act: Create the configuration from the charm.
+    assert: Correct configurations from the mock charm.
+    """
+    charm = MockCharmFactory()
+    charm.config[PROXY_CACHE_VALID_CONFIG_NAME] = '["200 302 30m", "400 1m", "500 1m"]'
+
+    config = Configuration.from_charm(charm)
+    assert config.hostname == "example.com"
+    assert config.path == "/"
+    assert config.backends == (IPv4Address("10.10.1.1"), IPv4Address("10.10.2.2"))
+    assert config.protocol == "https"
+    assert config.health_check_interval == 30
+    assert config.backends_path == "/"
+    assert config.proxy_cache_valid == ("200 302 30m", "400 1m", "500 1m")
```

Comment:
Do you think these could just be parametrized tests that expect valid and invalid values? ALso make sure we check edge cases thoroughly

Reply:
I have parametrized the tests and added more test cases with the valid values.